assets/js/job-editor/JobEditorMain.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react"
+import JobEditor from "./JobEditor"
+import type { WithActionProps } from "../react/lib/with-props";
+import { sortMetadata } from "../metadata-loader/metadata";
+import { useWorkflowStore } from "../workflow-editor/store";
+import pDebounce from "p-debounce";
+import { EDITOR_DEBOUNCE_MS } from "../common";
+import type { Lightning } from "#/workflow-diagram/types";
+
+interface JobEditorComponentProps {
+  job_id: string,
+  adaptor: string,
+  source: string,
+  disabled: boolean,
+  disabled_message: string,
+}
+
+export const JobEditorMain: WithActionProps<JobEditorComponentProps> = (props) => {
+  const [metadata, setMetadata] = React.useState<false | object>(false);
+  const [source, setSource] = React.useState('');
+
+  // use workflowstore here
+  const { change, getById } = useWorkflowStore();
+
+  // debounce editor content update
+  const _debouncedPushChange = pDebounce((content: string) => {
+    change({ jobs: [{ id: props.job_id, body: content }] })
+  }, EDITOR_DEBOUNCE_MS);
+
+  // init hook - getting metadata
+  React.useEffect(() => {
+    props.handleEvent('metadata_ready', (payload) => {
```

Comment:
Maybe out of scope but I'd love to see a central definition of all the events and their expected payloads somewhere. Can we fix it or raise an issue to do it soon?